### PR TITLE
Introduce repack module

### PR DESCRIPTION
**Pull request by axic**
Inspired by https://github.com/ewasm/sentinel-rs/pull/7 from @cdetrio.

### CODE REVIEW

**Comment by axic** [comment]
Need to finish the test cases.

File: libchisel/src/repack.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            .build()
+            .build()
+            .export()
+            .field("_call")
+            .internal()
+            .func(0)
+            .build()
+            .build();
+
+        let trimmer = TrimExports::with_preset("pwasm").unwrap();
```
Comment (jakelang) [inline libchisel/src/repack.rs]: what's going on here with these tests?
From what I gather this should be repack.

File: README.md [hunk 2]
```diff
@@ -43,6 +43,10 @@ Wraps module into an ewasm-compatible constructor. It has two presets:
 - `memory`: wrap the module as a pre-defined memory section
 - `customsection`: include the module as a custom section
 
+### repack
+
+Re-serializes the module. It will drop any unknown (custom) sections.
```
Comment (jakelang) [inline README.md]: I think the wording in the first sentence is a bit unintuitive. Maybe the second sentence on its own will do.

File: libchisel/src/repack.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+impl ModuleTranslator for Repack {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Err(ModuleError::NotSupported)
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let module = module.clone();
+        let module = builder::from_module(module).build();
```
Comment (jakelang) [inline libchisel/src/repack.rs]: Are we certain that this drops custom sections?

File: libchisel/src/repack.rs (hunk 1 above)
Comment (axic) [reply libchisel/src/repack.rs]: As I said above tests need to be finished. The source file was trimexports.

File: libchisel/src/repack.rs (hunk 3 above)
Comment (axic) [reply libchisel/src/repack.rs]: Based on empirical tests it drops everything.

**Comment by axic** [comment]
> Maybe would be better to include this as an option for DropSection?

Yeah when dropsection is finished, but this works already?

File: libchisel/src/repack.rs (hunk 3 above)
Comment (jakelang) [reply libchisel/src/repack.rs]: everything being all "unknown" sections?

Comment (axic) [reply libchisel/src/repack.rs]: It dropped the names section too. Technically it just drops every unparsed section and none of the custom sections are parsed by default by pwasm.

File: README.md (hunk 2 above)
Comment (axic) [reply README.md]: Well the second sentence is not only what it does, it is a side effect.

**Comment by axic** [comment]
Merging this now as it is a need in for other projects (scout, ewasm-precompiles).

**Pull request merged**